grpc-gcp/src/main/java/com/google/cloud/grpc/metrics/GcpMetrics.java

```diff
@@ -0,0 +1,436 @@
+package com.google.cloud.grpc.metrics;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The GcpMetrics class provides a structured way to define and configure metrics. It defines
+ * standard attributes, predefined metrics, metric types, units of measurement, and a flexible
+ * configuration system using builders. This framework helps ensure consistency and ease of use when
+ * instrumenting GCP API clients.
+ */
+public class GcpMetrics {
+  // Convenient map of all built-in attributes to their names.
+  public static final Map<Attr, String> allBuiltInAttributes =
+      Collections.unmodifiableMap(
+          Stream.of(Attr.values()).collect(Collectors.toMap((Attr x) -> x, Objects::toString)));
+
+  // All predefined metrics.
+  public static final Set<MetricConfig> allMetrics =
+      Collections.unmodifiableSet(
+          new HashSet<>(
+              Stream.of(Metric.values())
+                  .map(ConfigurableMetric::getDefaultConfig)
+                  .collect(Collectors.toSet())));
+
+  /**
+   * The Attr enum defines a set of standard attributes that can be associated with metrics. These
+   * attributes provide additional context about the metric being recorded. For example, STATUS
+   * represents the gRPC status code, METHOD represents the gRPC method name, and so on.
+   */
+  public enum Attr {
+    STATUS("grpc.status"),
+    METHOD("grpc.method"),
+    METHOD_TYPE("grpcgcp.method_type"),
+    TRANSPARENT_RETRY("grpcgcp.tr_retry");
+
+    private final String attr;
+
+    Attr(String attr) {
+      this.attr = attr;
+    }
+
+    @Override
+    public String toString() {
+      return attr;
+    }
+  }
+
+  /**
+   * The Metric enum defines a set of predefined metrics that can be recorded. Each metric has a
+   * Kind (e.g., COUNTER, HISTOGRAM), an internal Unit (e.g., NANOSECONDS, OCCURENCES), and a
+   * default MetricConfig.
+   */
+  public enum Metric implements ConfigurableMetric {
```

Comment:
Could you help to check https://github.com/grpc/grpc-java/tree/master/opentelemetry and re-architect the implementation in a similar way?
We don't need to define our own model which forks the opentelemetry concepts.